### ISSUE TITLE
sort by confidence level when identifying multiple compounds

### ISSUE THREAD

**Issue by dylanhross** [dylanhross/lipydomics#11]
In the various `id_feat_...` functions, multiple identifications can be yielded for a given search criterion. Currently, they are only added in the order of their appearance when searching the database. There should be a way of ranking them on the basis of their relative confidence levels (_i.e._ how close are the query values to the theoretical values). This can be done using a score derived from either the *l1* or *l2* norm of the residuals vector. Example:
- consider a 3-component match using m/z (`mz_q`) retention time (`rt_q`) and CCS (`ccs_q`)
- `q` is the query vector defined as `q = <mz_q, rt_q, ccs_q>`
- `x` is one potential feature vector returned from the query, defined as `x = <mz_x, rt_x, ccs_x>`
- a residual vector `r` can be computed as follows `r = <mz_r, rt_r, ccs_r> = <mz_x - mz_q, rt_x - rt_q, ccs_x - ccs_q>`
- We could directly take the *l1* or *l2* norm of this vector to get a confidence score, but it would not be very informative due to the widely different scales and precisions of each value. 
- One way to address this issue would be to apply a simple set of empirically determined _rough_ scaling factors to the residual vector, based on knowledge about the scale and precision that can be expected from each value: `r_adj = <100, 10, 1> * r = <100 * mz_r, 10 * rt_r, 1 * ccs_r>`
- A related but slightly different approach would be to derive the scaling factors from the already specified tolerance vector (`tol = <mz_tol, rt_tol, ccs_tol>`), since these also reflect the scale and precision of each value. In this case, we compute an adjusted residual vector as follows: `r_adj = r / tol = <mz_r / mz_tol, rt_r / rt_tol, ccs_r/ccs_tol> = <mz_ra, rt_ra, ccs_ra>`
- finally, we can take either the *l1* or *l2* norm of the adjusted residuals to derive a score that is inversely proportional to confidence in a given prediction: (*l1*) `1/score = abs(mz_ra) + abs(rt_ra) + abs(ccs_ra)` (*l2*) `1/score = sqrt(mz_ra^2 + rt_ra^2 + ccs_ra^2)`
- The choice of which norm to use will depend upon whether similar residuals in all components or especially small residuals in some components (and acceptance of some larger residuals in others) is more desirable. 

**In summary**,  need to associate some type of scoring with the identifications based upon how close the observed values are to queried values so that multiple identifications can be compared to one another.
